doc/elements.md

```diff
@@ -1,16 +1,93 @@
-
-# Elements for use in `question.html`
+# PrairieLearn Elements for use in `question.html`
+
+When writing questions, there exists a core pool of elements that provides 
+common structures associated with assessment items. These elements can be
+split into two distinct groups: **submission** and **decorative**. Within this
+document, all of PrairieLearn's elements are displayed alongside links to sample
+elements within the example course.
+
+**Submission** elements act as a way to receive a response or input from the
+student. These elements are traditionally referred to as form input fields. 
+PrairieLearn presently provides the following templated **input field** elements:
+
+- [`pl-multiple-choice`](#pl-multiple-choice-element): Selecting only 
+  **one option** from a list.
+- [`pl-checkbox`](#pl-checkbox-element): Selecting **multiple options** from a
+  list.
+- [`pl-number-input`](#pl-number-input-element): Fill in a **numerical** value 
+  within a specific tolerance level such as 3.14, -1.921, and so on.
+- [`pl-integer-input`](#pl-integer-input-element): Fill in an **integer** value 
+  such as -71, 0, 5, 21, and so on.
+- [`pl-symbolic-input`](#pl-symbolic-input-element): Fill in a **symbolic** value
+  such as `x^2`, `sin(z)`, `mc^2`, and so on.
+- [`pl-string-input`](#pl-string-input-element): Fill in a **string** value
+  such as "Illinois", "GATTACA", "computer", and so on.
+- [`pl-matrix-component-input`](#pl-matrix-component-input-element): Fill in
+  a **matrix** using grid that has an input area for each element.
+- [`pl-matrix-input`](#pl-matrix-input-element): Supply a matrix in a supported
+  programming language format.
+- [`pl-file-editor`](#pl-file-editor-element): Provide an in-browser code editor
+  for writing and submitting code.
+- [`pl-file-upload`](#pl-file-upload-element): Provide a submission area
+  to obtain a file with a specific naming scheme.
+- [`pl-threejs`](#pl-threejs-element): Enables 3D scene display and problem
+  submission.
+
+**Decorative** elements are meant to improve how the question is displayed to
+students. Elements under this category include ways to specify question markup,
+images, files, and code display. The following **decorative** elements are available:
+
+- [`pl-code`](#pl-code-element): Displays code rendered with the appropriate
+  syntax highlighting. 
+- [`pl-figure`](#pl-figure-element): Embed an image file in the question.
+- [`pl-file-download`](#pl-file-download-element): Enable file downloads for
+  data-centric questions.
+- [`pl-variable-output`](#pl-variable-output-element): Displays matrices in 
+  code form for supported programming languages.
+- [`pl-matrix-latex`](#pl-matrix-latex-element): Displays matrices using 
+  appropriate LaTeX commands for use in a mathematical expression.
+- [`pl-question-panel`](#pl-question-panel-element): Displays the text of a question.
```

Comment:
Consider moving these to a **conditional** section? They don't so much display the text of a question as they do conditionally render their children depending on panel; I think that distinction is worth making clear and expanding on.

Reply:
Agreed. I've rearranged the document to account for this.

Though, I think another round (not this PR) should probably add a section to the documentation that emphasizes how one can use these w.r.t custom elements.